pkg/scalers/azure_pipelines_scaler.go

```diff
@@ -374,7 +385,9 @@ func getCanAgentDemandFulfilJob(jr JobRequest, metadata *azurePipelinesMetadata)
 			}
 		}
 	}
-
+	if metadata.requireAllDemands {
+		return countDemands == len(demandsAvail) && len(demandsAvail) == len(demandsReq)-1
+	}
 	return countDemands == len(demandsReq)-1
```

Comment:
```suggestion
	matchDemands := countDemands == len(demandsReq)-1
	if metadata.requireAllDemands {
		return matchDemands && countDemands == len(demandsAvail)
	}
	return matchDemands
```

With this change, we clarify in the code that in both cases we require the demands matching, and one case is a superset of the other.
WDYT?